### PR TITLE
[KOGITO-5195] - Support optimistic locking for concurrency control in…

### PR DESCRIPTION
**Pull request by swati-kale**
… infinispan, mongodb, postgres and jdbc with quarkus and springboot examples.
Added seperate profile - persistenceWithLock.

Signed-off-by: Swati Kale <swkale@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `master` branch!**

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by swati-kale** [comment]
@cristianonicolai please review.

**Comment by cristianonicolai** [comment]
ok to test

File: process-infinispan-persistence-quarkus/src/main/resources/application.properties [hunk 1]
```diff
@@ -4,6 +4,7 @@
 quarkus.swagger-ui.always-include=true
 
 quarkus.infinispan-client.server-list=localhost:11222
+kogito.persistence.optimistic.lock=${enable.lock}
```
Comment (cristianonicolai) [inline process-infinispan-persistence-quarkus/src/main/resources/application.properties]:
```suggestion
kogito.persistence.optimistic.lock=${enable.lock:false}
```

File: process-infinispan-persistence-quarkus/pom.xml [hunk 2]
```diff
@@ -77,5 +77,22 @@
         </executions>
       </plugin>
     </plugins>
+    <resources>
```
Comment (cristianonicolai) [inline process-infinispan-persistence-quarkus/pom.xml]: @swati-kale with the change below to the application.properties, you can use system properties to configure the locking, that way you can avoid filtering the resources folder.

File: process-mongodb-persistence-quarkus/pom.xml [hunk 3]
```diff
@@ -112,5 +122,11 @@
         </dependency>
       </dependencies>
     </profile>
+    <profile>
+      <id>persistenceWithLock</id>
+      <properties>
+        <enable.lock>true</enable.lock>
```
Comment (tiagodolphine) [inline process-mongodb-persistence-quarkus/pom.xml]: can't we just do `<kogito.persistence.optimistic.lock>true</kogito.persistence.optimistic.lock>`?
in this way, we would avoid putting this property in the application.properties, wdyt? just trying to simplify...

**Review by tiagodolphine — COMMENTED** [review]
LGTM in general, I just put a comment, to try to simplify the configuration.

**Comment by tiagodolphine** [comment]
@MarianMacik can you take a look at this, thanks !!

File: process-infinispan-persistence-quarkus/pom.xml (hunk 2 above)
Comment (swati-kale) [reply process-infinispan-persistence-quarkus/pom.xml]: @cristianonicolai I did the changes suggested by you, but they are not working.

File: process-mongodb-persistence-quarkus/pom.xml (hunk 3 above)
Comment (swati-kale) [reply process-mongodb-persistence-quarkus/pom.xml]: @tiagodolphine I removed the kogito.persistence.optimistic.lock from application.properties and tried using <kogito.persistence.optimistic.lock>true</kogito.persistence.optimistic.lock>, but it doesnot work.

Comment (cristianonicolai) [reply process-mongodb-persistence-quarkus/pom.xml]: +1, this should work as
`java  -Dkogito.persistence.optimistic.lock=true -jar target/quarkus-app/quarkus-run.jar` or
`mvn clean compile quarkus:dev -Dkogito.persistence.optimistic.lock=true`

Comment (swati-kale) [reply process-mongodb-persistence-quarkus/pom.xml]: I was told to create seperate profile for lock in the meeting, so chose this  approach.
Runtime properties through command line is certainly easy option.  @cristianonicolai, I will do necessary changes as we discussed.

**Review by cristianonicolai — APPROVED** [review]

**Comment by cristianonicolai** [comment]
@tiagodolphine looks good to you?

**Review by tiagodolphine — APPROVED** [review]
LGTM, thanks @swati-kale

**Pull request merged**